### DISCUSSION TITLE
Add Temporary fly

### DISCUSSION THREAD

**Discussion by MeoNguOfficial** [EssentialsX/Essentials#5793] [Ideas and Feature Suggestions] [answered]
I hope we'll add Temporary fly on next version 2.21.0

**Comment by Chew** [comment]
Can you elaborate on your use-case and why this should be added?

**Reply by MeoNguOfficial** [reply]
Even though there is a flight feature, there is no flight time limit, meaning permanent flight, so that's why I propose adding a temporary flight feature.

**Reply by comehelpmae** [reply, marked as answer]
You can do this with most permission plugins by temp adding a permission, and than having a check if they have it, if not they get their fly taken away if they were in the process of using it

**Reply by MeoNguOfficial** [reply]
Hmm i have a question: I set temporary by a temporary permission, player can reactive flight it (example, a pkayer can fly on 30 minutes, and continue fly, type /fly again)?

**Reply by Evidentsinger14** [reply]
EssentialsX will not automatically disable their flight when they no longer have the permission. You can however enable the setting to reset their flight on world change, which will check for the permission. If you need something to automatically disable it after X amount of time, i'd suggest looking for a temp flight type of plugin, EssentialsX only provides a basic implementation of flight.

**Reply by MeoNguOfficial** [reply]
hmm so hard
